lib/src/common/cache/cache.dart

```diff
@@ -294,6 +301,66 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
     return new Future.value();
   }
 
+  /// Provides access to a value stored in the cache without informing the
+  /// [CachingStrategy] of a get.
+  ///
+  /// [weakGet] can be thought of sort of like a weak reference when working with
```

Comment:
I think the term "weak" here is misleading (I'm going to be PRing a `WeakMap` implementation soon that uses true weak references).

Reply:
If there's going to be a `WeakMap` I agree :/ this name will be confusing. `getIfCached` isn't bad maybe `getExisting`?